torch/csrc/distributed/c10d/CUDASymmetricMemoryUtils.cpp

```diff
@@ -0,0 +1,223 @@
+#include <sys/socket.h>
+#include <sys/syscall.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#include <c10/util/error.h>
+
+#if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
+#include <c10/cuda/driver_api.h>
+#endif
+
+#include <torch/csrc/distributed/c10d/CUDASymmetricMemoryUtils.hpp>
+#include <torch/csrc/distributed/c10d/Store.hpp>
+#include <torch/csrc/distributed/c10d/cuda/utils.hpp>
+
+namespace c10d::symmetric_memory {
+
+bool device_has_multicast_support(int device_idx) {
+  if (c10::utils::check_env("TORCH_SYMM_MEM_DISABLE_MULTICAST") == true) {
```

Comment:
planning to document these ENVs? any idea if they are going to be long-lived?

Reply:
We can probably clean this env.